text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,181 @@
+# Propagate head trace sampling probability
+
+Use the W3C trace context to convey consistent head trace sampling probability.
+
+## Motivation
+
+The head trace sampling probability is the probability associated with
+the start of a trace context that determines whether child contexts
+are sampled or not when using the `ParentBased` Sampler.  It is useful
+to know the head trace sampling probability associated with a context
+in order to build span-to-metrics pipelines when the built-in
+`ParentBased` Sampler is used.
+
+A consistent trace sampling decision is one that can be carried out at
+any node in a trace, which supports collecting partial traces.
+OpenTelemetry specifies a built-in `TraceIDRatioBased` Sampler that
+aims to accomplish this goal but was left incomplete (see
+[TODOs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#traceidratiobased) in the specification).
+
+We propose to propagate the necessary information alongside the [W3C
+sampled flag](https://www.w3.org/TR/trace-context/#sampled-flag) using
+`tracestate` with an `otelprob` vendor tag.
+
+## Explanation
+
+Two pieces of information are needed to convey consistent head trace
+sampling probability:
+
+1. The head trace sampling probability
+2. Source of consistent sampling decisions.
+
+This proposal uses one byte of information for each of these.
+
+### Probability value
+
+To limit the cost of this extension and for statistical reasons
+documented below, we propose to limit head trace sampling probability
+to powers of two.  This limits the available head trace sampling
+probabilities to 1/2, 1/4, 1/8, and so on.  We can compactly encode
+these probabilities as small integer values using the base-2 logarithm
+of the adjusted count (i.e., inverse probability).
+
+For example, the probability value 2 corresponds with 1-in-4 sampling,
+the probability value 10 corresponds with 1-in-1024 sampling.  Using
+one byte of information we can convey sampling rates as small as 2^-255.
+
+### Random value
+
+With head trace sampling probabilities limited to powers of two, the
+amount of randomness needed per trace context is limited.  A
+consistent sampling decision is accomplished by propagating a
+geometrically distributed random variable with shape parameter `1/2`,
+requiring only two bits of randomness on average per trace.  See
```

Comment:
not following this

Reply:
I've rewritten this passage, hoping to clarify.
```
A
consistent sampling decision is accomplished by propagating a specific
random variable.  The random variable is a described by a discrete
geometric distribution having shape parameter `1/2`, listed below:

| Value | Probability |
| ----- | ----------- |
| 0 | 1/2 |
| 1 | 1/4 |
| 2 | 1/8 |
| 3 | 1/16 |
| 4 | 1/32 |
| ... | ... |
| N | 1/(2^(N+1)) |

See [Estimation from Partially Sampled Distributed
Traces](https://arxiv.org/pdf/2107.07703.pdf) section 2.8 for a
detailed explanation.

Such a random variable `r` can be generated using the following
pseudocode:

```
r := 0
for {
  if nextRandomBit() {
    break // The expected value of r is 2
  }
  r++
}
```
```